rustls/src/common_state.rs

```diff
@@ -51,6 +51,66 @@ pub struct CommonState {
     pub(crate) enable_secret_extraction: bool,
 }
 
+#[cfg(feature = "std")]
+impl CommonState {
```

Comment:
Style nit: I like the idea of splitting the `impl`, but I think the guarded `impl` should go directly below the unguarded one.

(Same for other split `impl`s.)